src/frontend/apps/desk/src/features/auth/useAuthStore.tsx

```diff
@@ -1,40 +1,47 @@
 import { create } from 'zustand';
 
-import { initKeycloak } from './keycloak';
+import { UserData, getMe } from '@/features/auth/api';
 
 interface AuthStore {
   authenticated: boolean;
   initAuth: () => void;
   initialized: boolean;
   logout: () => void;
-  token: string | null;
+  userData?: UserData;
 }
 
 const initialState = {
   authenticated: false,
   initialized: false,
-  token: null,
+  userData: undefined,
 };
 
 export const useAuthStore = create<AuthStore>((set) => ({
   authenticated: initialState.authenticated,
   initialized: initialState.initialized,
-  token: initialState.token,
+  userData: initialState.userData,
 
   initAuth: () =>
     set((state) => {
-      if (process.env.NEXT_PUBLIC_KEYCLOAK_LOGIN && !state.initialized) {
-        initKeycloak((token) => set({ authenticated: true, token }));
-        return { initialized: true };
+      if (state.initialized) {
+        return {};
       }
-
-      /**
-       * TODO: Implement OIDC production authentication
-       */
-
+      getMe()
+        .then((data: UserData) => {
+          set({ initialized: true, authenticated: true, userData: data });
+          return { initialized: true };
+        })
+        .catch(() => {
+          // todo - implement a proper login screen to prevent automatic navigation.
```

Comment:
FMI, Agent connect doesn't manage the login screen ?

Reply:
Sorry, I mean the first screen on the Figma. There is a landing page with the Agent Connect button at the top right. I can update my comment.

Reply:
<img width="604" alt="Capture d’écran 2024-02-16 à 11 05 42" src="https://github.com/numerique-gouv/people/assets/45729124/f47c05f7-7c97-4a1a-b24c-9a724ee0c7ec">